webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -145,24 +145,30 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
     this.initForm();
   }
 
-  private initForm() {
+  private async initForm() {
     this.contentError = false;
     this.errorTranslationKey = false;
 
-    return this
-      .getContact()
-      .then((contact) => this.getForm(contact))
-      .then((formId) => this.renderForm(formId))
-      .then((formInstance) => this.setEnketoContact(formInstance))
-      .then(() => {
-        this.globalActions.setLoadingContent(false);
-      })
-      .catch((err) => {
-        this.errorTranslationKey = err.translationKey || 'error.loading.form';
-        this.globalActions.setLoadingContent(false);
-        this.contentError = true;
-        console.error('Error loading contact form.', err);
-      });
+    try {
+      const contact = await this.getContact();
+      const contactTypeId = this.contactTypesService.getTypeId(contact) || this.routeSnapshot.params?.type;
+      const contactType = await this.contactTypesService.get(contactTypeId);
+      if (!contactType) {
+        console.error(`Unknown contact type "${contactTypeId}"`);
```

Comment:
We should throw an error here instead of logging it! 
Please have a look at what happens when you try to load a form for a contact type that doesn't exist, then make this code throw an error and see the difference.

Reply:
Wow, thanks for pointing this out! Although this is the same as the old behavior, the new behavior from throwing an error provides a much better UX.

Reply:
Previously, some other code (getForm) threw an error. But now we just "return" if we don't find the contactType before that code is reached.